### PR TITLE
Add some support for formulae with functions in defined names

### PR DESCRIPTION
**Pull request by cyrelk**
Hi Guyon,

This is my first pull request on github, so hope it is correct :)

Anyway as described in my issue this is a preliminary support for defined names that have functions in their formulae. Without this patch the functions names are treated as worksheet names.

Kind Regards
Cyril

### CODE REVIEW

**Comment by cyrelk** [comment]
This is a start, for example if both formula and range are void the name is returned nevertheless. 
The code support only formula with functions inside, what do if we have simple operators?

File: spec/unit/doc/defined-names.spec.js [hunk 1]
```diff
@@ -70,15 +70,17 @@ describe('DefinedNames', function() {
   it('skips values with invalid range', function() {
     var dn = new DefinedNames();
     dn.model = [
-      {name: 'eq', ranges: ['"="']},
-      {name: 'ref', ranges: ['#REF!']},
-      {name: 'single', ranges: ['Sheet3!$A$1']},
-      {name: 'range', ranges: ['Sheet3!$A$2:$F$2228']}
+      {name: 'eq', ranges: ['"="'], formula: ''},
```
Comment (Siemienik) [inline spec/unit/doc/defined-names.spec.js]: input data in tests should not be modified, it causes lost possibility to test backward compatibility

File: spec/unit/xlsx/xform/book/defined-name-xform.spec.js [hunk 2]
```diff
@@ -8,17 +8,17 @@ var expectations = [
   {
     title: 'Defined Names',
     create: function() { return new DefinedNameXform(); },
-    preparedModel: {name: 'foo', ranges: ['bar!$A$1:$C$1']},
+    preparedModel: {name: 'foo', ranges: ['bar!$A$1:$C$1'], formula: ''},
```
Comment (Siemienik) [inline spec/unit/xlsx/xform/book/defined-name-xform.spec.js]: input data in tests should not be modified, it causes lost possibility to test backward compatibility

File: spec/unit/xlsx/xform/book/defined-name-xform.spec.js [hunk 3]
```diff
@@ -8,17 +8,17 @@ var expectations = [
...
     xml: '<definedName name="foo">bar!$A$1:$C$1</definedName>',
-    parsedModel: {name: 'foo', ranges: ['bar!$A$1:$C$1']},
+    parsedModel: {name: 'foo', ranges: ['bar!$A$1:$C$1'], formula: ''},
     tests: ['render', 'renderIn', 'parse']
   },
   {
     title: 'Print Area',
     create: function() { return new DefinedNameXform(); },
-    preparedModel: {name: '_xlnm.Print_Area', localSheetId: 0, ranges: ['bar!$A$1:$C$10']},
+    preparedModel: {name: '_xlnm.Print_Area', localSheetId: 0, ranges: ['bar!$A$1:$C$10'], formula: ''},
```
Comment (Siemienik) [inline spec/unit/xlsx/xform/book/defined-name-xform.spec.js]: input data in tests should not be modified, it causes lost possibility to test backward compatibility

**Comment by alubbe** [comment]
@cyrelk just wanted to check if you're still available to work on this?

**Comment by cyrelk** [comment]
@alubbe 

Not anymore, sorry. The project I'm working on is full office.js now.

**Pull request closed without merging**